### PR TITLE
[FSDP][Docs] Document get_state_dict_type

### PR DESCRIPTION
**Pull request by rohan-varma**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #102890
* #102692
* #102680
* #102672
* __->__ #102658
* #102657
* #102639
* #102637
* #102508

Per title

Differential Revision: [D46335317](https://our.internmc.facebook.com/intern/diff/D46335317/)

### CODE REVIEW

**Review by fegin — APPROVED** [review]
Nice!

File: torch/distributed/fsdp/fully_sharded_data_parallel.py [hunk 1]
```diff
@@ -702,6 +702,17 @@ def set_state_dict_type(
 
     @staticmethod
     def get_state_dict_type(module: nn.Module) -> StateDictSettings:
+        """
+        Get the state_dict_type and the corresponding configurations
+        for the FSDP modules rooted at ``module``. The target module
+        does not have to be an FSDP module. Note that this method will
+        throw if the ``StateDictSettings`` for differenet FSDP submodules
```
Comment (awgu) [inline torch/distributed/fsdp/fully_sharded_data_parallel.py:709]:
```suggestion
        throw if the ``StateDictSettings`` for different FSDP submodules
```

File: torch/distributed/fsdp/fully_sharded_data_parallel.py [hunk 2]
```diff
@@ -702,6 +702,17 @@ def set_state_dict_type(
...
+        differ.
+
+        Returns:
+            A ``StateDictSettings`` containing the state_dict_type and
+            state_dict / optim_state_dict configs that are currently set.
```
Comment (awgu) [inline torch/distributed/fsdp/fully_sharded_data_parallel.py:714]: nit: Same with "state_dict_type", "state_dict", and "optim_state_dict" here.

File: torch/distributed/fsdp/fully_sharded_data_parallel.py [hunk 3]
```diff
@@ -702,6 +702,17 @@ def set_state_dict_type(
 
     @staticmethod
     def get_state_dict_type(module: nn.Module) -> StateDictSettings:
+        """
+        Get the state_dict_type and the corresponding configurations
```
Comment (awgu) [inline torch/distributed/fsdp/fully_sharded_data_parallel.py:706]: nit: We should use double backtick for "state_dict_type" or use "state dict type" without underscores.

File: torch/distributed/fsdp/fully_sharded_data_parallel.py [hunk 4]
```diff
@@ -702,6 +702,17 @@ def set_state_dict_type(
...
+        for the FSDP modules rooted at ``module``. The target module
+        does not have to be an FSDP module. Note that this method will
```
Comment (awgu) [inline torch/distributed/fsdp/fully_sharded_data_parallel.py:708]: nit: Should we just document via `Raises:` in the docstring the condition for raising an error?

**Review by awgu — APPROVED** [review]
Stamp to unblock

File: torch/distributed/fsdp/fully_sharded_data_parallel.py (hunk 3 above)
Comment (rohan-varma) [reply torch/distributed/fsdp/fully_sharded_data_parallel.py:706]: Will just leave them as non-backticked

**Pull request closed without merging**